elasticsearch_query.py

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env python
+
+# Copyright 2014, Rackspace US, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import datetime
+import json
+import optparse
+import os
+import re
+import requests
+import sys
+
+from maas_common import metric, status_ok, status_err, print_output
+
+
+def get_elasticsearch_bind_host():
+    """Get the bindhost for elasticsearch if we can read the config."""
+    config = '/etc/elasticsearch/elasticsearch.yml'
+    if not os.path.exists(config):
+        return 'localhost'
+
+    with open(config) as fd:
+        contents = fd.readlines()
+    bind_host_re = re.compile('^network\.bind_host:\s+([0-9\.]+)\s+')
+    hosts = filter(bind_host_re.match, contents)
+    if not hosts:
+        raise SystemExit(False)
+    match = bind_host_re.match(hosts[-1])
+    return match.groups()[0]
+
+ES_PORT = '9200'
+ELASTICSEARCH = None
+
+
+def json_querystring(query_string, sort=None):
+    """Generate the JSON data for a query_string query."""
+    query_dict = {'query': {'query_string': query_string}}
+    if sort:
+        query_dict['sort'] = sort
+    return json.dumps(query_dict)
+
+
+def json_filter(filtered_query):
+    """Generate the JSON data for a filtered query."""
+    return json.dumps({'query': {'filtered': filtered_query}})
+
+
+def search_url_for(index=None):
+    """Generate the search URL for an index."""
+    if index is not None:
+        return ELASTICSEARCH + '/' + index + '/_search'
+    return ELASTICSEARCH + '/_search'
+
+def count_url_for(index=None):
+    """Generate the search URL for an index."""
+    if index is not None:
+        return ELASTICSEARCH + '/' + index + '/_count'
+    return ELASTICSEARCH + '/_count'
+
+
+def get_json(url, data):
+    """Wrap calls to requests to handle exceptions."""
+    exceptions = (requests.exceptions.HTTPError,
+                  requests.exceptions.ConnectionError)
+    try:
+        r = requests.get(url, data=data)
```

Comment:
Does Elasticsearch return JSON even during a 404 or 411 or something else? If not you might want to add (to this try block)

```py
r.raise_for_status()
```

Which will raise an HTTPError.